gradle/plugins/aot-smoke-test-plugin/src/main/java/org/springframework/aot/gradle/AotSmokeTestPlugin.java

```diff
@@ -89,6 +91,33 @@ private void configureBootJavaProject(Project project) {
 				.mavenLocal(
 						(mavenLocal) -> mavenLocal.content((content) -> includedGroups.forEach(content::includeGroup)));
 		}
+		if (project.hasProperty("forceSnapshots")) {
+			project.getConfigurations().all((configuration) ->
+					configuration.getResolutionStrategy().eachDependency(dependency -> {
+						boolean isSnapshot = dependency.getRequested().getVersion() != null
+								&& dependency.getRequested().getVersion().endsWith("-SNAPSHOT");
+						if (dependency.getRequested().getGroup().startsWith("org.springframework") && !isSnapshot) {
+							String[] versionParts = dependency.getRequested().getVersion().split("-");
+							String snapshotVersion = versionParts[0] + "-SNAPSHOT";
```

Comment:
This looks great. Thanks.

I wonder if it needs to be tweaked a little for maintenance branches. For example, if Boot's using Spring Security `6.1.5` then should forcing snapshots should move to `6.1.6-SNAPSHOT`? I think it probably should otherwise we won't be testing the latest changes.

Reply:
Yes, we probably need to account for that. I wonder if when Spring Boot 3.1.x moves to SNAPSHOTs, the tests will fail because we will try to resolve a patch version that still does not exist.

Reply:
I think your existing logic that leaves things as they are when the version is already a `-SNAPSHOT` will avoid that problem.

I think we need the following:

- `x.y.z-SNAPSHOT` is left as-is
- `x.y.z` is changed to `x.y.z+1-SNAPSHOT`
- `x.y.0-Mz` is changed to `x.y.0-SNAPSHOT`
- `x.y.0-RCz` is changed to `x.y.0-SNAPSHOT`